name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/advanziabank/AdvanziaBankPDFExtractorTest.java

```diff
@@ -0,0 +1,72 @@
+import static name.abuchen.portfolio.datatransfer.ExtractorMatchers.interest;
+import static name.abuchen.portfolio.datatransfer.ExtractorMatchers.removal;
+import static name.abuchen.portfolio.datatransfer.ExtractorTestUtilities.countAccountTransactions;
+import static name.abuchen.portfolio.datatransfer.ExtractorTestUtilities.countBuySell;
+import static name.abuchen.portfolio.datatransfer.ExtractorTestUtilities.countSecurities;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.collection.IsEmptyCollection.empty;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Test;
+
+import name.abuchen.portfolio.datatransfer.Extractor.Item;
+import name.abuchen.portfolio.datatransfer.actions.AssertImportActions;
+import name.abuchen.portfolio.datatransfer.pdf.AdvanziaBankPDFExtractor;
+import name.abuchen.portfolio.datatransfer.pdf.PDFInputFile;
+import name.abuchen.portfolio.model.Client;
+import name.abuchen.portfolio.money.CurrencyUnit;
+
+@SuppressWarnings("nls")
+public class AdvanziaBankPDFExtractorTest
+{
+    @Test
+    public void testKontoauszug01()
+    {
+        AdvanziaBankPDFExtractor extractor = new AdvanziaBankPDFExtractor(new Client());
+
+        List<Exception> errors = new ArrayList<>();
+
+        List<Item> results = extractor.extract(PDFInputFile.loadTestCase(getClass(), "Kontoauszug01.txt"), errors);
+
+        assertThat(errors, empty());
+        assertThat(countSecurities(results), is(0L));
+        assertThat(countBuySell(results), is(0L));
+        assertThat(countAccountTransactions(results), is(6L));
+        assertThat(results.size(), is(6));
+        new AssertImportActions().check(results, CurrencyUnit.EUR);
+
+        // assert transaction
+        assertThat(results, hasItem(deposit(hasDate("2023-06-14"), hasAmount("EUR", 100), //
+                        hasSource("Kontoauszug01.txt"))));
+
+        // assert transaction
+        assertThat(results, hasItem(removal(hasDate("2023-06-15"), hasAmount("EUR", 100), //
+                        hasSource("Kontoauszug01.txt"))));
+
+        // assert transaction
+        assertThat(results, hasItem(deposit(hasDate("2023-06-16"), hasAmount("EUR", 5000), //
+                        hasSource("Kontoauszug01.txt"))));
+
+        // assert transaction
+        assertThat(results, hasItem(deposit(hasDate("2023-06-20"), hasAmount("EUR", 9200), //
+                        hasSource("Kontoauszug01.txt"))));
+
+        // assert transaction
+        assertThat(results, hasItem(deposit(hasDate("2023-06-30"), hasAmount("EUR", 3000), //
+                        hasSource("Kontoauszug01.txt"))));
+
+        // assert transaction
+        assertThat(results, hasItem(interest(hasDate("2023-06-30"), hasAmount("EUR", 47.11), //
+                        hasSource("Kontoauszug01.txt"))));
```

Comment:
Add `hasNote(null)` to the tests so that all possibilities are checked completely.

Reply:
Has been added